app/webpacker/components/EditUser/api/getUserDetails.js

```diff
@@ -0,0 +1,18 @@
+import { fetchJsonOrError } from '../../../lib/requests/fetchWithAuthenticityToken';
+import { viewUrls } from '../../../lib/requests/routes.js.erb';
+
+export default async function getUserDetails(userId) {
+  const { data } = await fetchJsonOrError(
+    viewUrls.users.showForEdit(userId),
+  );
+
+  const userDetails = {
+    id: data.id,
+    name: data.name,
+    dob: data.dob,
+    gender: data.gender,
+    country_iso2: data.country_iso2,
+  };
```

Comment:
If your intention is to create an exact copy (as far as I can tell, all keys that you are creating are exactly identical to `data`, then you can also just do `{ ...data }`.

Reply:
But also: Why are you creating a shallow copy in the first place, instead of directly returning `data`?

Reply:
Basically I don't want the full data to go to the frontend because remaining data are not necessary there (especially it's very huge I assume). I don't see any negative effects for sending the `data` completely as well. Should I send it as `data` instead?

Reply:
If you don't want the full data to go to the frontend (ie you don't need it at all) then consider creating a dedicated endpoint which only gives you the fields that you need. Bonus: This avoids redundant database queries for stuff like the user avatar, which is quite costly to retrieve.